### PR TITLE
GO-4442 add weights

### PR DESCRIPTION
**Pull request by fat-fellow**
---

- [ ] I understand that contributing to this repository will require me to agree with the [CLA](https://github.com/anyproto/.github/blob/main/docs/CLA.md)

---

### What type of PR is this? (check all applicable)

- [ ] 🍕 Feature
- [ ] 🐛 Bug Fix
- [ ] 📝 Documentation Update
- [ ] 🎨 Style
- [ ] 🧑‍💻 Code Refactor
- [ ] 🔥 Performance Improvements
- [ ] ✅ Test
- [ ] 🤖 Build
- [ ] 🔁 CI

### Added tests?

- [ ] 👍 yes
- [ ] 🙅 no, because they aren't needed
- [ ] 🙋 no, because I need help

### Added to documentation?

- [ ] 📜 README.md
- [ ] 📓 [tech-docs](https://github.com/anyproto/tech-docs)
- [ ] 🙅 no documentation needed

### CODE REVIEW

File: pkg/lib/localstore/ftsearch/autobatchertantivy.go [hunk 1]
```diff
@@ -5,10 +5,17 @@ import (
...
 )
 
-const docLimit = 10000
+type AutoBatcher interface {
+	// UpdateDoc adds a update operation to the batcher. If the batch is reaching the size limit, it will be indexed and reset.
+	UpdateDoc(doc SearchDoc) error
+	// DeleteDoc adds a delete operation to the batcher
+	// maxSize limit check is not performed for this operation
+	DeleteDoc(id string) error
+	// Finish performs the
```
Comment (jmetrikat) [inline pkg/lib/localstore/ftsearch/autobatchertantivy.go:16]: what's the missing part of the comment?

File: pkg/lib/localstore/ftsearch/autobatchertantivy.go [hunk 2]
```diff
@@ -5,10 +5,17 @@ import (
...
+	Finish() error
```
Comment (jmetrikat) [inline pkg/lib/localstore/ftsearch/autobatchertantivy.go:17]: what's the missing part of the comment?

File: pkg/lib/localstore/objectstore/spaceindex/queries.go [hunk 3]
```diff
@@ -288,78 +280,29 @@ func (s *dsObjectStore) performQuery(q database.Query) (records []database.Recor
...
 		}
 
 		resultsByObjectId[path.ObjectId] = append(resultsByObjectId[path.ObjectId], result)
 	}
 
 	for objectId := range resultsByObjectId {
-		sort.Slice(resultsByObjectId[objectId], func(i, j int) bool {
-			if bleveResults[i].Score > bleveResults[j].Score {
+		cur := resultsByObjectId[objectId]
+		sort.Slice(cur, func(i, j int) bool {
```
Comment (cheggaaa) [inline pkg/lib/localstore/objectstore/spaceindex/queries.go:304]: slices.SortFunc should be faster

File: pkg/lib/localstore/objectstore/spaceindex/queries.go [hunk 4]
```diff
@@ -419,6 +364,37 @@ func (s *dsObjectStore) performFulltextSearch(text string, highlightFormatter ft
 	return results, nil
 }
 
+func convertToHighlightRanges(ranges [][]int, highlight string) []*model.Range {
+	var highlightRanges []*model.Range
+
+	byteToRuneIndex := make([]int, len(highlight)+1)
+	for i := range byteToRuneIndex {
+		byteToRuneIndex[i] = utf8.RuneCountInString(highlight[:i])
```
Comment (cheggaaa) [inline pkg/lib/localstore/objectstore/spaceindex/queries.go:372]: you should work with utf16 (and please, check it in other fts places)

Comment (fat-fellow) [reply pkg/lib/localstore/objectstore/spaceindex/queries.go:372]: I've found in payments

Comment (fat-fellow) [reply pkg/lib/localstore/objectstore/spaceindex/queries.go]: it's been reverted in payments

**Review by deff7 — APPROVED** [review]

**Pull request merged**